src/Compat/CompatForm.php

```diff
@@ -22,4 +47,28 @@ public function hasDefaultElementDecorator()
 
         return true;
     }
+
+    /**
+     * Return a duplicate of the given submit button with the `class` attribute fixed to `primary-submit-btn-duplicate`
+     *
+     * @param FormSubmitElement $originalSubmitButton
+     *
+     * @return SubmitElement
+     */
+    public function duplicateSubmitButton(FormSubmitElement $originalSubmitButton): SubmitElement
+    {
+        $attributes = array_reduce(
```

Comment:
I can't imagine a valid reason for using `array_reduce` here.

Though, I guess the intention was to have the same attributes on the duplicate as on the original? This fails miserably because a) `array_reduce` returns a single result if used correctly and b) the current incorrect usage returns an empty array, always.

Though, even if corrected, what's the actual reason for transferring the attributes anyway? The duplicate just needs the correct name and value. Anything else is not required, or is it? A problem I didn't noticed previously is also that the ID attribute is duplicated in this case, but duplicate IDs are invalid in HTML.

Reply:
The submit button to be cloned can have attributes and callbacks that must also be present on the clone, such as `formaction`, `formmethod` and `formnovalidate`, so simply not using the original attributes is not an option.

With Icinga/ipl-html#110 merged, I suggest somethin like the following code here:

```php
// Until proper cloning (ref here) is implemented ...
$attributes = clone $originalSubmitButton->getAttributes();
$submit = new SubmitElement($originalSubmitButton->getName(), $attributes);
$attributes->registerAttributeCallback('name', [$submit, 'getNameAttribute']);

Reply:
I've added a test for additional attributes to be carried over. I also fixed the `array_reduce` - I was missing a return.

Regarding the ID, how do you think this should be handled? Either an appended name or a unique name would work.

Reply:
Attribute cloning is now merged.